frontend/src/UseFormSender.js

```diff
@@ -0,0 +1,31 @@
+import { useState } from 'react';
+import axios from 'axios';
+
+const UseFormSender = (reqType, url) => {
+    const [error, setError] = useState(null);
+    const [loading, setLoading] = useState(false);
+    const [data, setData] = useState(null);
+
+    const cb = async (requestBody) => {
+        setLoading(true);
+        setError(null);
+        try {
+            const response = await axios({
+                method: reqType,
+                url,
+                headers: {
+                    'Content-Type': 'application/json'
+                },
+                data: requestBody
+            });
```

Comment:
Please create an axios instance with default overrides for your app. Create a new file e.g. `src/axios.js` create there a new axios instance with overriden defaults e.g. `'Content-Type': 'application/json'` already set and if possible also set the request prefix like `http://localhost:8080/` in there. I think for creating a new axios instance you need `axios.create` but you need to check the docs.

Also have in mind that if this app is deployed to a website requests  with `http://localhost:8080/` won't work as the front end code is fetched to the user's laptop and `http://localhost:8080/` will the the localhost of that specific user. To be able to connect to your own backend you'll need to use a rather domain name of your website or it's ip or if the backend and frontend are hosted on the same place you can use your current pages url. (Those should be changed depending on an env variable, yes FE also has env variabes, they are accessed using process.env.NAME_OF_THE_ENV_VAR and it I'm right create react app only allows you to set custom env vars with REACT_APP prefix and they are embedded to the front end build during `npm run build`)

Reply:
This might be too much info. Please ask questions to me or Gago if what :)